tensorrt_llm/commands/serve.py

```
@@ -864,6 +867,14 @@ def launch_visual_gen_server(
     # uvicorn. VisualGen initialization can take many minutes; if we deferred
     # the bind until uvicorn started, anything else on the host could grab the
     # port in that window and trtllm-serve would die at bind() time.
+    # port == 0 lets the kernel pick the port; the caller then needs a way
+    # to learn it, either by service discovery or by report_addr. Validate
+    # before getaddrinfo, which rejects negative ports with its own error.
+    if not (port > 0 or disagg_cluster_config is not None or report_addr):
```

Ruff (0.16.2)

[error] Undefined name `disagg_cluster_config`

(F821)